frontend/src/pages/create_campaign/Preview/index.js

```diff
@@ -0,0 +1,87 @@
+import React, { useState, useEffect } from "react";
+import PropTypes from "prop-types";
+import { SubmitButton, SubmitWrapper } from "./campaignPreview.styled";
+import ApplicationForm from "../../../components/ApplicationForm";
+
+// FIXME: CHAOS-65, add campaign card to preview tab
+const CampaignReview = (props) => {
+  const {
+    value,
+    index,
+    questions,
+    roles,
+    campaignName,
+    headerImage,
+    description,
+    onSubmit,
+  } = props;
+  const [rolesSelected, setRolesSelected] = useState([]);
+  const [answers, setAnswers] = useState({});
+  useEffect(() => {
+    questions.forEach((q) => {
+      if (!(q.id in answers)) {
+        answers[q.id] = "";
```

Comment:
OOO careful - you're altering a state directly which is bad practice and cause undefined behaviour

i would create a new variable `newAnswers` or something similar and then set all my `newAnswers[q.id]` there

even better, since you're creating a new variable, try use declarative instead of imperative style, so your code should look something like

```
const newAnswers = answers
   .filter((question) => (insert_valid_qid_logic_here))
   .map((question) => (insert_initialising_logic_here));
   
setAnswers(newAnswers);
```